sources/packages/backend/libs/queue/src/queue.root.module.ts

```diff
@@ -0,0 +1,55 @@
+import { Module } from "@nestjs/common";
+import { BullModule, BullRootModuleOptions } from "@nestjs/bull";
+import Redis, { Cluster, RedisOptions } from "ioredis";
+import { QUEUE_PREFIX } from "./constants/queue.constant";
+
+/**
+ * Creates root connection to redis standalone or redis cluster
+ * based on environment variable.
+ */
+@Module({
+  imports: [
+    BullModule.forRootAsync({
+      useFactory: getConnectionFactory,
+    }),
+  ],
+})
+export class QueueRootModule {}
+
+/**
+ * Connection factory which returns connection properties
+ * to connect redis.
+ * Depending upon the environment variable it uses standalone
+ * or cluster connection.
+ ** While running in local env make sure to set REDIS_STANDALONE_MODE as true.
+ * @returns redis connection factory
+ */
+function getConnectionFactory():
+  | Promise<BullRootModuleOptions>
+  | BullRootModuleOptions {
+  const redisConnectionOptions: RedisOptions = {
+    host: process.env.REDIS_HOST || "localhost",
+    port: +process.env.REDIS_PORT || 6379,
+    password: process.env.REDIS_PASSWORD,
+  };
+  if (process.env.REDIS_STANDALONE_MODE === "true") {
+    return {
+      redis: redisConnectionOptions,
+      prefix: QUEUE_PREFIX,
+    };
+  }
+  return {
```

Comment:
so, this scenario will not happen right.? as `process.env.REDIS_STANDALONE_MODE === "true"` will always true

Reply:
`REDIS_STANDALONE_MODE === "true"` will happen only on DEV local machine. Once deployed to OpenShift it will always use the cluster mode because the ENV variable is not even being created there.

Reply:
oh..okay. thanks for that 👍